runtime/interpreter/value.go

```diff
@@ -8672,7 +8672,9 @@ func (v *CompositeValue) Transfer(
 
 	dictionary := v.dictionary
 
-	needsStoreTo := v.NeedsStoreTo(address)
+	currentAddress := v.StorageID().Address
+
+	needsStoreTo := address != currentAddress
```

Comment:
is this different to `v.NeedsStoreTo(address)`?

Reply:
No, I just "inlined" it here, to avoid the recomputation in `StorageID()` to get the current address